src/api_client.rs

```diff
@@ -0,0 +1,200 @@
+    };
+
+    GQLClient::new_with_config(ClientConfig {
+        endpoint: api_url,
+        timeout: Some(10),
+        headers: Some(headers),
+        proxy: None,
+    })
+}
+
+nest! {
+    #[derive(Debug, Deserialize, Serialize)]*
+    #[serde(rename_all = "camelCase")]*
+    struct CreateLoginSessionData {
+        create_login_session: pub struct CreateLoginSessionPayload {
+            pub callback_url: String,
+            pub session_id: String,
+        }
+    }
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct ConsumeLoginSessionVars {
+    session_id: String,
+}
+nest! {
+    #[derive(Debug, Deserialize, Serialize)]*
+    #[serde(rename_all = "camelCase")]*
+    struct ConsumeLoginSessionData {
+        consume_login_session: pub struct ConsumeLoginSessionPayload {
+            pub token: Option<String>
+        }
+    }
+}
+
+#[derive(Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct FetchLocalRunReportVars {
+    pub owner: String,
+    pub name: String,
+    pub run_id: String,
+}
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+pub enum ReportConclusion {
+    AcknowledgedFailure,
+    Failure,
+    MissingBaseRun,
+    Success,
+}
+
+impl Display for ReportConclusion {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ReportConclusion::AcknowledgedFailure => {
+                write!(f, "{}", style("Acknowledged Failure").yellow().bold())
+            }
+            ReportConclusion::Failure => write!(f, "{}", style("Failure").red().bold()),
+            ReportConclusion::MissingBaseRun => {
+                write!(f, "{}", style("Missing Base Run").yellow().bold())
+            }
+            ReportConclusion::Success => write!(f, "{}", style("Success").green().bold()),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FetchLocalRunReportHeadReport {
+    pub id: String,
+    pub impact: Option<f64>,
+    pub conclusion: ReportConclusion,
+}
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum RunStatus {
+    Pending,
+    Processing,
+    Completed,
+}
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FetchLocalRunReportRun {
+    pub id: String,
+    pub status: RunStatus,
+    pub head_reports: Vec<FetchLocalRunReportHeadReport>,
+}
+nest! {
+    #[derive(Debug, Deserialize, Serialize)]*
+    #[serde(rename_all = "camelCase")]*
+    struct FetchLocalRunReportData {
+        repository: pub struct FetchLocalRunReportRepository {
+            settings: struct FetchLocalRunReportSettings {
+                allowed_regression: f64,
+            },
+            runs: Vec<FetchLocalRunReportRun>,
+        }
+    }
+}
+
+pub struct FetchLocalRunReportResponse {
+    pub allowed_regression: f64,
+    pub run: FetchLocalRunReportRun,
+}
```

Comment:
It is a struct I recreated to avoid deep nesting, it is manually instantiated in the code, not being used to parse the response from the API.
In the original `FetchLocalRunReportData` struct, the `allowed_regression` is in `response.repository.settings.allowed_regression`